recipes/geotrans/all/test_package/conanfile.py

```diff
@@ -0,0 +1,23 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class GeotransTestConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = (
+        "cmake",
+        "cmake_find_package",
+    )
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self):
+            with tools.environment_append(
+                {"MSPCCS_DATA": self.deps_user_info["geotrans"].data_path}
+            ):
+                bin_path = os.path.join("bin", "example")
+                self.run(bin_path, run_environment=True)
```

Comment:
```suggestion
            bin_path = os.path.join("bin", "example")
            self.run(bin_path, run_environment=True)
```

Reply:
The idea behind this was to be explicit about illustrating to the consumer that the `MSPCCS_DATA` environment variable _must_ be set prior to running a program that imports the library as per the discussion in https://github.com/conan-io/conan-center-index/issues/6511. The same applies to the changes you've suggested to remove the information output. This isn't an obvious requirement of the library, and by setting the environment variable in the package recipe I'm concerned that it obfuscates what's going on from anybody looking to the `test_package` consumer recipe for an exemplar of how to use the library. Additionally, This environment variable won't be persistent beyond the build execution, but it will be required to be set at runtime of the consumer executable - I think it might be clearer if we keep this part as is?

Reply:
I disagree:

- `env_info` is here for these kind of environment variables. 
- should we provide some advices (and I don't think that recipe is the proper place, there should be library documentation), build() is not the good location because it's not always called.
- it prevents any consumer using virtualrunenv generator to run application out of the box.

Reply:
1. Not opposed to `env_info` provided it's acceptable to use the consumer conanfile to provide exemplar usage of binary execution with the environment variable with comments
2. I was using `user_info` at the suggestion of @madebr - it's neither here nor there for me, the problem it solves was demonstrating to the user that an environment variable needed to be set when executing the binary, and that the data path could be obtained at least from within the consumer conanfile
3. I think if there's anything we can do to ease the user experience we definitely should. It wasn't clear to me that this environment variable was a requirement. The library documentation is verbose and this isn't obvious in it. Is there a more appropriate place for an output message than `build()`? My thought was that this seemed to be what would be executed most frequently. See 4.1 - 4.3 for info this should confer.
4. That sucks. Do you have any other suggestions on how we might be able to demonstrate the intended usage of the library? Noting that there is likely to be a point at which the executable is run independently of conan, at which time the user will need to know:
    1. That an environment variable needs to be set so the library can access its data files
    2. Where the data files have been installed to
    3. How the data files can be moved to a location of their choosing if they want to (even if this is just a link to the `imports()` method documentation)

What are your thoughts?

**CONTEXT**
The library makes some assumptions about the path of the data dir, specifically it assumes it lies in the relative path `../../data` with respect to the current working directory. Since that's not something we can impose on the consumer, we're required to use their backup method, which is the specification of the environment variable at runtime. The documentation does provide some information about this, but it finding it would require a detailed reading of their 185 page user guide or 35 page programmers guide. It wasn't clear to me that this was a requirement when writing the package, and I think if we can make the package easier to use we definitely should.

I think that the consumer conanfile is a good place for this because it demonstrates the `example` program being executed as it might be from a command line.This is a natural place to look for a usage example, and the output information can easily get buried in the wall of text that's output when building/installing dependencies so this serves to ensure that the message isn't lost. A commented example may make an acceptable substitute in lieu of functional code if this breaks the virtualrunenv generator. This might even be clearer because I'd be able to strip away all of the python specific syntax

Reply:
I've moved the output text to a comment-based commentary in the consumer conanfile.py. I've also accepted your change to use `env_info` rather than `user_info`. I've noticed that it's still required to append this to the environment prior to running the test example. I'm not sure which part of the above prevents the use of the virtualrunenv generator, but hopefully it was just the use of `user_info` instead of `env_info`. If it's the fact that the example binary was appending to the environment, could you please suggest a better solution than what I've amended the consumer conanfile to use?

Reply:
The problem with using `self.env_info` is that when using 2 profile builds, these variables aren't applied to the test package.